### PR TITLE
Use env_vars to for reload config test.

### PR DESCRIPTION
**Pull request by jhtimmins**
Updates the test logging configuration to use the `env_vars` to set the environment variables instead of setting them explicitly within the `settings_context` context manager.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: tests/test_logging_config.py [hunk 1]
```diff
@@ -128,43 +128,41 @@ def settings_context(content, directory=None, name='LOGGING_CONFIG'):
...
-                open(os.path.join(basedir, '__init__.py'), 'w').close()
-                basedir = os.path.join(basedir, part)
+    # initial_logging_config = os.environ.get("AIRFLOW__LOGGING__LOGGING_CONFIG_CLASS", "")
+    # try:
+    settings_root = tempfile.mkdtemp()
+    filename = f"{SETTINGS_DEFAULT_NAME}.py"
+    if directory:
+        # Replace slashes by dots
+        module = directory.replace('/', '.') + '.' + SETTINGS_DEFAULT_NAME + '.' + name
+        # breakpoint()
```
Comment (mik-laj) [inline tests/test_logging_config.py]: Can you remove it?

File: tests/test_logging_config.py [hunk 2]
```diff
@@ -128,43 +128,41 @@ def settings_context(content, directory=None, name='LOGGING_CONFIG'):
...
-            pathlib.Path(dir_path).mkdir(parents=True, exist_ok=True)
-
-            # Add the __init__ for the directories
-            # This is required for Python 2.7
-            basedir = settings_root
-            for part in directory.split('/'):
-                open(os.path.join(basedir, '__init__.py'), 'w').close()
-                basedir = os.path.join(basedir, part)
+    # initial_logging_config = os.environ.get("AIRFLOW__LOGGING__LOGGING_CONFIG_CLASS", "")
+    # try:
```
Comment (mik-laj) [inline tests/test_logging_config.py]: Can you remove it?

File: tests/test_logging_config.py [hunk 3]
```diff
@@ -128,43 +128,41 @@ def settings_context(content, directory=None, name='LOGGING_CONFIG'):
...
+        open(os.path.join(basedir, '__init__.py'), 'w').close()
 
-            settings_file = os.path.join(dir_path, filename)
-        else:
-            module = SETTINGS_DEFAULT_NAME + '.' + name
-            settings_file = os.path.join(settings_root, filename)
+        settings_file = os.path.join(dir_path, filename)
+    else:
+        module = SETTINGS_DEFAULT_NAME + '.' + name
+        # breakpoint()
```
Comment (mik-laj) [inline tests/test_logging_config.py]: Can you remove it?

File: tests/test_logging_config.py (hunk 1 above)
Comment (jhtimmins) [reply tests/test_logging_config.py]: Oops, my bad

File: tests/utils/test_dag_processing.py [hunk 4]
```diff
@@ -377,34 +377,35 @@ class path, thus when reloading logging module the airflow.processor_manager
...
-            processor_agent.start()
-            if not async_mode:
-                processor_agent.run_single_parsing_loop()
-
-            processor_agent._process.join()
-            # Since we are reloading logging config not creating this file,
-            # we should expect it to be nonexistent.
-
-            self.assertFalse(os.path.isfile(log_file_loc))
+            with env_vars({('logging', 'logging_config_class'): 'custom_airflow_local_settings.LOGGING_CONFIG'}):
```
Comment (mik-laj) [inline tests/utils/test_dag_processing.py]:
```
tests/utils/test_dag_processing.py:380:0: C0301: Line too long (113/110) (line-too-long)
```

Comment (kaxil) [inline tests/utils/test_dag_processing.py]:
```suggestion
            with env_vars({
                ('logging', 'logging_config_class'): 'custom_airflow_local_settings.LOGGING_CONFIG'
            }):
```

**Review by kaxil — APPROVED** [review]
LGTM, just a minor suggestion to fix the failing static check

**Pull request closed without merging**